lib/src/pubspec.dart

```diff
@@ -138,15 +138,14 @@ class Pubspec {
   ///
   /// When [lenient] is set, top-level property-parsing or type cast errors are
   /// ignored and `null` values are returned.
-  factory Pubspec.parse(String yaml, {sourceUrl, bool lenient = false}) {
-    lenient ??= false;
-
-    return checkedYamlDecode(
-        yaml, (map) => Pubspec.fromJson(map, lenient: lenient),
-        sourceUrl: sourceUrl);
-  }
-
-  static List<String> _normalizeAuthors(String author, List<String> authors) {
+  factory Pubspec.parse(String yaml, {Uri? sourceUrl, bool lenient = false}) =>
+      checkedYamlDecode(
+        yaml,
+        (map) => Pubspec.fromJson(map!, lenient: lenient),
+        sourceUrl: sourceUrl,
+      );
+
+  static List<String> _normalizeAuthors(String? author, List<String>? authors) {
```

Comment:
Should this be checking that both are not non-null?

Reply:
author/authors is now deprecated -     https://dart.dev/tools/pub/pubspec#authorauthors


I'm going to leave the logic as-is for now and deprecate the fields/ctor params